### ISSUE TITLE
handling unexpected errors (with no ruleID)

### ISSUE THREAD

**Issue by alanbsmith** [appfolio/eslint-rails#11]
One thing I noticed is that if there's a warning created that doesn't have a specific `ruleID`, associated with it (random syntax errors for example), the linter blows up with this error: `NoMethodError: undefined method `size' for nil:NilClass`, which isn't super helpful.

That is happening because ruleID is never being set here:
https://github.com/appfolio/eslint-rails/blob/master/lib/eslint-rails/warning.rb#L9

A quick fix would be something like:
```
      @rule_id = warning_hash['ruleId'] || "-1"
      # OR
     @rule_id = warning_has['ruleId'] || "unexpected error"
```

This would allow the linter to finish and put out an error like this:
`4:8       high   -1 Parsing error: Unexpected token <`

Which is a little more helpful (IMO).

I'll submit a PR shortly if that's how you'd like to go about it. Let me know, and I'm happy to modify it however you'd like.

Cheers and thanks again for the tool!

Alan

**Comment by alanbsmith** [comment]
Suggesting a PR for this here: https://github.com/appfolio/eslint-rails/pull/13
